libtransmission/net.h

```diff
@@ -104,13 +111,70 @@ constexpr bool tr_address_is_valid(tr_address const* a)
  * Sockets
  **********************************************************************/
 
-/* https://en.wikipedia.org/wiki/Differentiated_services#Class_Selector */
-enum
-{
-    TR_IPTOS_LOWCOST = 0x38, /* AF13: low prio, high drop */
-    TR_IPTOS_LOWDELAY = 0x70, /* AF32: high prio, mid drop */
-    TR_IPTOS_THRUPUT = 0x20, /* CS1: low prio, undef drop */
-    TR_IPTOS_RELIABLE = 0x28 /* AF11: low prio, low drop */
+/*
+ * Definitions for DiffServ Codepoints as per RFCs 2474, 3246, 4594 & 8622.
+ * Not all are guaranteed to be defined in <netinet/ip.h> so we keep a copy of the DSCP values.
+ */
+#ifndef IPTOS_DSCP_AF11
+#define IPTOS_DSCP_AF11  0x28
+#define IPTOS_DSCP_AF12  0x30
+#define IPTOS_DSCP_AF13  0x38
+#define IPTOS_DSCP_AF21  0x48
+#define IPTOS_DSCP_AF22  0x50
+#define IPTOS_DSCP_AF23  0x58
+#define IPTOS_DSCP_AF31  0x68
+#define IPTOS_DSCP_AF32  0x70
+#define IPTOS_DSCP_AF33  0x78
+#define IPTOS_DSCP_AF41  0x88
+#define IPTOS_DSCP_AF42  0x90
+#define IPTOS_DSCP_AF43  0x98
+#define IPTOS_DSCP_EF  0xb8
+#endif
+
+#ifndef IPTOS_DSCP_CS0
+#define IPTOS_DSCP_CS0  0x00
+#define IPTOS_DSCP_CS1  0x20
+#define IPTOS_DSCP_CS2  0x40
+#define IPTOS_DSCP_CS3  0x60
+#define IPTOS_DSCP_CS4  0x80
+#define IPTOS_DSCP_CS5  0xa0
+#define IPTOS_DSCP_CS6  0xc0
+#define IPTOS_DSCP_CS7  0xe0
+#endif
```

Comment:
If the DSCP_AF_ and DSCP_CS values are known and never going to change, maybe just make them constexpr ints e.g. `static constexpr int TR_IPTOS_DSCP_CS4 = 0x80;` so that we never have to worry about header conflicts?